packages/contentstack-auth/src/commands/auth/login.ts

```
@@ -98,22 +93,19 @@ export default class LoginCommand extends BaseCommand<typeof LoginCommand> {
     }
   }
 
-  async login(username: string, password: string, totpSecret?: string): Promise<void> {
+  async login(username: string, password: string): Promise<void> {
     log.debug('Starting login process', { ...this.contextDetails, username });
 
     try {
       log.debug('Calling auth handler login', this.contextDetails);
       let tfaToken: string | undefined;
       
-      if (totpSecret) {
-        log.debug('Generating TOTP token from provided secret', this.contextDetails);
-        try {
-          tfaToken = totpHandler.generateTOTPFromSecret(totpSecret);
-          log.debug('TOTP token generated successfully', this.contextDetails);
-        } catch (error) {
-          log.debug('Failed to generate TOTP token', { ...this.contextDetails, error });
-          tfaToken = undefined;
-        }
+      try {
+        tfaToken = await totpHandler.getTOTPCode();
+        log.debug('TOTP token generated from stored configuration', this.contextDetails);
+      } catch (error) {
+        log.debug('Failed to generate TOTP token from config', { ...this.contextDetails, error });
+        tfaToken = undefined;
```

@cs-raj , we can throw the error in case Failed to generate TOTP token.